torch/nn/parallel/data_parallel.py

```diff
@@ -55,9 +55,10 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
 
     def forward(self, *inputs, **kwargs):
         inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
-        if len(self.device_ids) == 1:
+        n_gpus = len(self.device_ids)
+        if n_gpus == 1:
             return self.module(*inputs[0], **kwargs[0])
-        replicas = self.replicate(self.module, self.device_ids[:len(inputs)])
+        replicas = self.replicate(self.module, self.device_ids[:n_gpus])
```

Comment:
I don't think it's correct. The `self.device_ids[:len(self.device_ids)]` is always equal to just `self.device_ids`. We've been using `len(inputs)` because if you e.g. gave a batch of size 3 to DataParallel using 4 GPUs, then `scatter` will return fewer results, and this slicing is necessary.